store/key_store.go

```diff
@@ -52,12 +52,7 @@ func (ks *KeyStore) Unlock(phrase string) error {
 }
 
 // SignTx uses the unlocked account to sign the given transaction.
-func (ks *KeyStore) SignTx(tx *types.Transaction, chainID uint64) (*types.Transaction, error) {
-	account, err := ks.GetAccount()
-	if err != nil {
-		return nil, err
-	}
-
+func (ks *KeyStore) SignTx(account accounts.Account, tx *types.Transaction, chainID uint64) (*types.Transaction, error) {
```

Comment:
Just to reiterate what you've already mentioned, this is just a wrapper around `go-ethereum`'s existing method:

```golang
	return ks.KeyStore.SignTx(
		account,
		tx,
 		big.NewInt(int64(chainID)),
```

I see why it would be more intuitive as a method on the account object. As it stands now, `account` is a light type with just two immutable fields. I don't really see it as a child of keystore as much as a plain simple object, not a rich type. It seems that they have a mutex locking the **unlocked account** (different to account) on the keystore, which gives some justification to it living on keystore:

https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/keystore.go#L270-L285